plugin/storage/integration/integration.go

```diff
@@ -144,6 +150,34 @@ func (s *StorageIntegration) testGetServices(t *testing.T) {
 	}
 }
 
+func (s *StorageIntegration) testArchiveTrace(t *testing.T) {
+	defer s.CleanUp()
+	tID := model.NewTraceID(uint64(11), uint64(22))
+	expected := &model.Span{
+		OperationName: "archive_span",
+		StartTime:     time.Now().Add(-time.Hour * 72 * 5),
+		TraceID:       tID,
+		SpanID:        model.NewSpanID(55),
+		References:    []model.SpanRef{},
+		Process:       model.NewProcess("archived_service", model.KeyValues{}),
+	}
+	if s.SkipArchiveTest {
+		t.Skip("Skipping ArchiveTrace test because archive reader or writer is nil")
+	}
+	require.NoError(t, s.ArchiveSpanWriter.WriteSpan(context.Background(), expected))
+	s.Refresh()
+
+	var actual *model.Trace
+	found := s.waitForCondition(t, func(t *testing.T) bool {
+		var err error
+		actual, err = s.ArchiveSpanReader.GetTrace(context.Background(), tID)
+		return err == nil && len(actual.Spans) == 1
+	})
+	if !assert.True(t, found) {
+		CompareTraces(t, &model.Trace{Spans: []*model.Span{expected}}, actual)
+	}
```

Comment:
I suggest changing this to
```
require.True(t, found)
CompareTraces(...)
```
there's no need for the latter to be conditional, since require would abort the test anyway

Reply:
Noted

Reply:
The ES tests are failing after this change. Even if found is false, CompareTraces should only run if the found value is false right?

Reply:
if the value is not found the test failed

Reply:
Yes, the test should be skipped instead and compare traces shouldn't run, yes?

Reply:
no the test should. In the previous test `assert.True` was always called, so if that assert fails the whole test fails. It was just used to guard comparison which would create extra error noise if nothing was found. But if it's changed to require() then it simply doesn't reach that line. The test fails in either case.